File: src/main/java/com/eternalcode/plots/plot/PlotRepository.java
package com.eternalcode.plots.plot;

import java.util.UUID;
import java.util.concurrent.CompletableFuture;

public interface PlotRepository {

    CompletableFuture<Plot> loadPlot(UUID plotUUID);

    void savePlot(Plot plot);

    void deletePlot(Plot plot);

}

File: src/main/java/com/eternalcode/plots/plot/PlotRepository.kt
package com.eternalcode.plots.plot

import java.util.*
import java.util.concurrent.CompletableFuture

interface PlotRepository {
    fun loadPlot(plotUUID: UUID?): CompletableFuture<Plot?>?

    fun savePlot(plot: Plot?)

    fun deletePlot(plot: Plot?)
}
